components/suggest/src/db.rs

```diff
@@ -242,6 +247,7 @@ impl<'a> SuggestDao<'a> {
                                 ":suggestion_id": suggestion_id
                             },
                             |row| {
+                                if full_suffix.contains(keyword_suffix) {
```

Comment:
Sorry, I just noticed this: `.contains()` will match anywhere in the string, so `masking mail`, `masking i`, `masking il`, `masking l`, etc. will all match `masking email`.

Is that intentional? Or do we want this to be a prefix match, so only `masking e`, `masking em`, `masking ema`, etc. should match?

(Either way, let's add another test for that case, too, please—either a positive one if it's intentional, or a negative one that verifies we don't return anything. Thanks!)

Reply:
Side note: I also really like how you factored out `full_suffix` into a variable, and wondering if we can give the logic in the Pocket match arm the same treatment, while you're here?

```rs
let confidence = row.get("confidence")?;
let full_suffix = row.get::<_, String>("keyword_suffix")?;
let suffixes_match = match confidence {
    KeywordConfidence::Low => full_suffix.starts_with(keyword_suffix),
    KeywordConfidence::High => full_suffix == keyword_suffix,
};
```